### DISCUSSION TITLE
CreateDraftListing for digital product: "Error: A shipping_profile_id is required for physical listings"

### DISCUSSION THREAD

**Discussion by Wulfheart1** [etsy/open-api#1044] [Q&A]
As mentioned in the title i want to use the createDraftListing to create a digital product as draft listing.
The following screenshot shows the parameters i use for my test listing and also the api call in python:

![image](https://github.com/etsy/open-api/assets/137956992/24cd46bf-c1e0-4554-96f5-4b020d908780)

As response i get the error:
'error': 'A shipping_profile_id is required for physical listings.'

As my listing_type is set to 'download' and not to 'physical' this error looks like it is a bug. Hopefully you can help me here.

The used testing data:
{'quantity': 9, 'title': 'Abstrakte Fabelwesen Buntstift Zeichnung | Mischwesen Schakal Buntstift Poster | Moderne farbig-helle Zeichnung fürs Wohnzimmer', 'description': 'Tauche ein in eine Welt voller lebendiger Farben und dynamischer Formen mit dieser einzigartigen abstrakten Buntstiftzeichnung. &quot;Tanz des farbigen Schakals&quot; ist eine kraftvolle Darstellung der Schönheit und der Kreativität, die durch die Fusion von Farbe und Form entsteht.\n\nDiese handgefertigte Zeichnung wurde sorgfältig angefertigt. Jeder Strich und jede Linie ist sorgfältig platziert, um eine harmonische Balance zwischen den Farben und Formen zu schaffen. Die Kombination aus sanften Übergängen und kräftigen Kontrasten erzeugt eine faszinierende visuelle Dynamik, die den Betrachter in ihren Bann zieht.\n\nDie abstrakte Natur dieser Zeichnung ermöglicht es jedem Betrachter, seine eigene Interpretation zu finden und eine persönliche Verbindung zu den dargestellten Formen und Farben herzustellen. Obwohl die Komposition abstrakt ist, strahlt sie dennoch eine gewisse Lebendigkeit und Energie aus, die jedem Raum eine künstlerische Atmosphäre verleiht.\n\nDieses Kunstwerk eignet sich ideal als Blickfang in Wohnräumen, Büros oder anderen Innenräumen, die eine moderne und kreative Note erhalten möchten. Diese Zeichnung wird als digitales Produkt für den privaten Gebrauch verkauft, um dir die Freiheit zu geben, sie selbst zu drucken oder von einer professionellen Druckerei in deiner Nähe erstellen zu lassen. Dadurch ist es dir möglich sie entsprechend deinem eigenen Stil und deiner Vorliebe zu präsentieren. Beachte, dass kein pyhsisches Produkt geliefert wird. \n\nTechnische Details:\n\nTitel: &quot;Tanz des farbigen Schakals&quot;\n- Abstrakte Buntstiftzeichnung\n- Größe: 10752 x 16128 Pixel, 91.03 x 136.55 cm bei 300 DPI (hochauflösende Qualität)\n- Format: 2x3, 4x6, 6x9, 8x12, ...\n\nCopyright @ModernWatercolorLove. Kunstwerke dürfen nicht ohne Die Erlaubnis vervielfältigt oder weiterverkauft werden, bitte kontaktieren Sie mich über die Erlaubnis, wenn Sie benötigen. \n\nHilfeartikel zum Herunterladen eines digitalen Artikels:\nhttps://help.etsy.com/hc/en-us/articles/115013328108', 'price': 10.9, 'who_made': 'i_did', 'when_made': '2020_2023', 'taxonomy_id': 77, 'materials': [], 'tags': ['Schakal', 'digital', 'Wanddekoration', 'Zeichnung', 'boho', 'Skandinavisch', 'hell', 'buntstift', 'Fabelwesen', 'Wolf', 'Fledermaus', 'druckbare Kunst', 'abstrakt'], 'styles': [], 'is_personalizable': False, 'personalization_is_required': False, 'production_partner_ids': [], 'is_supply': False, 'is_customizable': False, 'should_auto_renew': False, 'is_taxable': True, 'listing_type': 'download'}

**Comment by Waterwitching** [comment]
I don't know if you solved this but I was getting the same error. To work around it I created a physical product and then converted it to digital afterwards using the update listing endpoint here - https://developer.etsy.com/documentation/reference/#operation/updateListing

````
# Convert the listing to a digital listing
            convert_to_digital_url = f'https://openapi.etsy.com/v3/application/shops/{shop_id}/listings/{new_listing_id}'

            convert_payload = {
                "type": "download"
            }

            response = requests.patch(convert_to_digital_url, headers=headers, json=convert_payload)
            if response.status_code == 200:
                print("Listing converted to digital successfully")
            else:
                print("Failed to convert listing to digital. Error:", response.text)
````

This page helped:

https://developer.etsy.com/documentation/tutorials/listings/#converting-a-physical-product-listing-to-a-digital-product-listing

but the "uploadlisting" endpoint doesn't exist in V3 & the instructions for creating a digital listing in the first place just kept returning the shipping_profile_id error.

**Reply by Wulfheart1** [reply]
Hi, thank you for this great answer. I have already tried different combinations of parameter values or removed/added different parameters in the body, but i was not able to get it working until now. i have already thought about your workaround, but first i have to integrate the necessary endpoints into my python client to be able to try it out. It would be great if it works this way.

According to the documentation the updateListing-endpoint should exist in v3, at least i would expect it as the url includes  "v3":
https://openapi.etsy.com/v3/application/shops/{shop_id}/listings/{listing_id}  
Why do you think it is not available in v3?

**Reply by Waterwitching** [reply]
Hi Wulfhear1, hope it helps, **_Updatelisting_** did work in the end to convert it to a digital listing, it was the **_uploadlisting_** endpoint that led me astray for a while. The page I shared says:

_"In the event that a physical product listing needs to be changed to a digital listing, this can be accomplished via the **[uploadListing]** (https://developer.etsy.com/documentation/reference/#operation/uploadListing) endpoint and passing type as "download". However, note that if the physical product listing has any variations or inventory beyond a single product, **updateListing** will return a 409 error."_

Once I realised it said **_updatelisting_** will return the error, I tried that endpoint & it all worked perfectly. Might be me, I was going cross-eyed by then but I couldn't find _**uploadlisting**_ as an endpoint by itself, there's only these:

uploadListingImage
uploadListingFile
uploadListingVideo

I don't know if it's confusion in the documentation with **_uploadListingFile_** - I'd also never say it's not me misunderstanding or reading it wrong either 😂

**Reply by amitkumar-248** [reply]
@Wulfheart1 plz for the Etsy tutorial link below it will surely help u 
[https://developers.etsy.com/documentation/tutorials/listings#listing-a-digital-product-for-sale](https://developers.etsy.com/documentation/tutorials/listings#listing-a-digital-product-for-sale)

thanks
